core/parachain/validator/impl/parachain_processor.cpp

```diff
@@ -3346,7 +3346,22 @@ namespace kagome::parachain {
           .proof = chunk->proof,
       };
     }
-    return network::FetchChunkResponse{};
+    return network::Empty{};
+  }
+
+  outcome::result<network::FetchChunkResponseObsolete>
+  ParachainProcessorImpl::OnFetchChunkRequestObsolete(
+      const network::FetchChunkRequest &request) {
+    if (auto chunk =
+            av_store_->getChunk(request.candidate, request.chunk_index)) {
+      if (chunk->index == request.chunk_index) {
```

Comment:
Why would `getChunk(index=index).index != index`?

Reply:
in rust `query_chunk -> AvailabilityStoreMessage::QueryChunk -> load_chunk`. Chunk index is equal `ValidatorIndex` and the response corresponds to request index.

Reply:
Both V1 and V2 have the same handler.

Reply:
https://github.com/paritytech/polkadot-sdk/blob/d2fd53645654d3b8e12cbf735b67b93078d70113/polkadot/node/core/av-store/src/lib.rs#L1345